File: 2021-08-05 (day18)/A_GO/solution2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> output;
        map<int, int> myMap;
        
        for(int i=0;i < nums.size();i++){
            myMap[nums[i]] = i;
        }
        for(int i=0;i < nums.size();i++){
            if(myMap[target-nums[i]] && myMap[target-nums[i]]!=i)
            {
                output.push_back(i);
                output.push_back(myMap[target-nums[i]]);
                break;
            }
        }
        return output;
    }
};

File: 2021-07/2021-07-23 (day5)/CXPhoenix/solution2.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        def searchHelper(l: int, r: int):
            mid = (r + l) // 2
            if r <= l:
                if target > nums[l]:
                    return l+1
                else:
                    return l
            if target > nums[mid]:
                return searchHelper(mid+1, r)
            else:
                return searchHelper(l, mid-1)
        return searchHelper(0, len(nums)-1)
        